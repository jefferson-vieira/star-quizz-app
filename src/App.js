import React from 'react';

import Router from './configs/router';

const App = () => <Router />;

export default App;
