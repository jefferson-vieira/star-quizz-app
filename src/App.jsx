import React from 'react';
import { connect } from 'react-redux';

import Loading from '@/components/Loading';
import Router from './routes/router';

const mapStateToProps = state => ({
  loading: state.status.loading
});

const App = connect(
  mapStateToProps,
  null
)(({ error, loading, refresh }) => (
  <>
    <div className="container-fluid">
      <Router />
    </div>
    <Loading show={loading} />
  </>
));

export default App;
