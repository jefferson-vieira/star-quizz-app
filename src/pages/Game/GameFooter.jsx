import React, { Component } from 'react';

class GameFooter extends Component {
  renderPages = () => {
    const { currentPage, totalItems, getPeople } = this.props;

    const pages = Math.ceil(totalItems / 10) || 0;

    const isActive = page => currentPage === page;

    return Array.from(new Array(pages), (val, index) => {
      const page = index + 1;
      return (
        <li key={page} className={`page-item ${isActive(page) && 'active'}`}>
          <a
            href="#game-header"
            className="page-link"
            title={`Ir para a p치gina ${page}`}
            onClick={() => getPeople(page)}
          >
            {page}
          </a>
        </li>
      );
    });
  };

  render() {
    const { currentPage, getPeople, next, previous } = this.props;

    return (
      <section id="game-footer" className="border-top border-warning pt-3">
        <nav aria-label="Game Navigation">
          <ul className="pagination pagination-lg justify-content-center">
            <li className={`page-item ${!previous && 'disabled'}`}>
              <a
                href="#game-body"
                className="page-link"
                title={`Ir para a p치gina ${currentPage - 1}`}
                onClick={() => getPeople(currentPage - 1)}
              >
                &lArr; Anterior
              </a>
            </li>
            {this.renderPages()}
            <li className={`page-item ${!next && 'disabled'}`}>
              <a
                href="#game-body"
                className="page-link"
                title={`Ir para a p치gina ${currentPage + 1}`}
                onClick={() => getPeople(currentPage + 1)}
              >
                Pr칩ximo &rArr;
              </a>
            </li>
          </ul>
        </nav>
      </section>
    );
  }
}

export default GameFooter;
