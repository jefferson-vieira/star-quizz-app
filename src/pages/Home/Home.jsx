import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import Logo from '@/components/Logo';
import Rules from '@/components/modals/Rules';

export default class Home extends Component {
  state = {
    showRules: false
  };

  handleRules = () => {
    this.setState(prevState => ({ showRules: !prevState.showRules }));
  };

  render() {
    return (
      <section id="home" className="home container my-3">
        <div className="row justify-content-center text-center">
          <Logo />
          <header id="home-title" className="home__title col-12">
            <h1>Bem-vindo ao Star Quizz App!</h1>
          </header>
          <section id="home-body" className="home__body col-12">
            <h2>
              Prove seus conhecimentos sobre{' '}
              <span className="home--yellow-text">Star Wars</span>.
            </h2>
            <h3>
              Para iniciar o desafio clique em{' '}
              <a href="#home-btn-play" className="home__play">
                jogar
              </a>
              .
            </h3>
            <h3>
              Mas lembre-se: ao clicar em JOGAR não haverá mais volta!
              <br />
              Está seguro disso? Então vá em frente e nos mostre...
            </h3>
            <div>
              <Link
                id="home-btn-rank"
                to="/rank"
                className="btn btn-outline-light btn-lg m-3"
                title="Mostrar pontuação"
              >
                Rank
              </Link>
              <button
                id="home-btn-play"
                type="button"
                className="btn btn-outline-primary btn-lg m-3"
                title="Mostrar regras"
                onClick={this.handleRules}
              >
                Jogar
              </button>
            </div>
          </section>
        </div>
        <Rules show={this.state.showRules} close={this.handleRules} />
      </section>
    );
  }
}
