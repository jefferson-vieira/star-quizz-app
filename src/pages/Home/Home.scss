@import 'assets/scss/variables';

.home {
  color: $white;

  &--yellow-text,
  &__title,
  &__play,
  &__play:hover {
    color: $yellow;
  }
}
