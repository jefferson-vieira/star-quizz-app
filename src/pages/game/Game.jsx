import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Header from './GameHeader';
import Body from './GameBody';

import { operations } from '../../store/ducks/game';

class Game extends Component {
  componentWillMount() {
    this.props.getPeople();
  }

  componentDidCatch() {
    alert('aff');
  }

  render() {
    return (
      <section id="game">
        <Header />
        <Body people={this.props.people} />
      </section>
    );
  }
}

const mapStateToProps = state => ({
  people: state.game.people
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      getPeople: operations.getPeople
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Game);
