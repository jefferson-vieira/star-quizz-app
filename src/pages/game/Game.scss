@import '../../assets/scss/breakpoints';
@import '../../assets/scss/variables';

.game {
  &-header {
    color: $yellow;

    &__timer {
      position: fixed;
      z-index: 10;
      top: 0;

      @include media-breakpoint-up(sm) {
        right: 0;
      }

      @include media-breakpoint-down(sm) {
        left: 1rem;
        right: 1rem;
      }

      & > div {
        background-color: rgba(0, 0, 0, 0.5);
      }

      & p {
        font-size: 2rem;
      }
    }
  }

  &-body {
    display: grid;
    grid-template-columns: 1fr;

    @include media-breakpoint-up(sm) {
      grid-template-columns: 1fr 1fr;
    }

    @include media-breakpoint-up(md) {
      grid-template-columns: 1fr 1fr 1fr;
    }

    @include media-breakpoint-up(lg) {
      grid-template-columns: 1fr 1fr 1fr 1fr;
    }

    @include media-breakpoint-up(xl) {
      grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
    }

    & .card {
      margin: 0.5rem;
    }
  }
}
