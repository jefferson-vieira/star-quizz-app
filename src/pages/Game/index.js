import './Game.scss';
import Game from './Game';

export default Game;
