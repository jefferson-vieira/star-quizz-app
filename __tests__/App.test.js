import React from 'react';
import { MemoryRouter } from 'react-router';

import { shallow, mount } from 'enzyme';

import App from '../src/App';

import Home from '../src/pages/home/Home';
import Game from '../src/pages/game/Game';

describe('Testing App Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Testing Router...', () => {
  it('should / refers to /home', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it('can redirect to /home', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/home']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Home)).toHaveLength(1);
  });

  it('can redirect to /game', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/game']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Game)).toHaveLength(1);
  });
});
