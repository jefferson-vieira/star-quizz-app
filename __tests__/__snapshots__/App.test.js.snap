// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Hello World!
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World!",
      },
      "ref": null,
      "rendered": "Hello World!",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hello World!
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World!",
        },
        "ref": null,
        "rendered": "Hello World!",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
